lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -3723,6 +3725,9 @@ getNetworkInformation nid NetworkLayer
                         Cardano.Testnet _ -> "testnet"
                     )
                     (fromIntegral $ unNetworkMagic $ toNetworkMagic nid)
+            , Api.walletMode = case lightSync of
+                Nothing -> Node
+                Just _ -> Light
```

Comment:
There is a data type that models a wallet's mode already and its named [`BlockchainSource`](https://github.com/input-output-hk/cardano-wallet/blob/light-mode/ADP-1663/expose-wallet-mode/lib/shelley/src/Cardano/Wallet/Shelley/BlockchainSource.hs):
```haskell
data BlockchainSource
    = NodeSource CardanoNodeConn NodeToClientVersionData SyncTolerance
    | BlockfrostSource Blockfrost.Project
```

This data is available in the `serveWallet` function https://github.com/input-output-hk/cardano-wallet/blob/light-mode/ADP-1663/expose-wallet-mode/lib/shelley/src/Cardano/Wallet/Shelley.hs#L175-L176

I suggest you to pass it as a parameter [here](https://github.com/input-output-hk/cardano-wallet/blob/ba1a99d3258a1893d8262b7bfdd3c19948068c47/lib/shelley/src/Cardano/Wallet/Shelley.hs#L323) and then use inside the `getNetworkInformation` function instead of recovering this information from the `NetworkLayer` itself. Alternatively, you can add the `BlockchainSource`  or maybe some `WalletMode` type derviced from it to the `NetworkLayer` as a field.